src/main/java/com/sprints/repository/SprintsCustomRepositoryImpl.java

```diff
@@ -28,5 +34,32 @@ public Sprint oneSprintBacklogValidation() {
 						.addCriteria(Criteria.where("is_backlog").is(true));
 		return mongoTemplate.findOne(query, Sprint.class);
 	}
-	
-}
+
+	@Override
+	public List<Sprint> findAllByParams(Optional<String> name, Optional<String> technology,
+			Optional<LocalDate> start_date, Optional<LocalDate> end_date) {
+		
+		Criteria criteria = new Criteria();
+		
+		if(name.isPresent()) {
```

Comment:
In the repository we should not have any if or validation. Le´st try to move all the IFs outside of the repository,

Reply:
Fixed